### PR TITLE
modify scan interface, unify fetch colname

### PR DESCRIPTION
**Pull request by critical27**
Some improvement in scan interface. Only one tag/edge is supported in one req. Add two flag, `only_latest_version` and `enable_read_from_follower`.

@Nicole00 @laura-ding , If want to set `only_latest_version` to true, `FLAGS_enable_multi_versions` should be true in storage, otherwise, performance will be hurt. By default `FLAGS_enable_multi_versions` is false, so just use the default value for `only_latest_version` is ok.

BTW, the column name in scan is still empty for now. It need to be unified with other interfaces in next PR. 

---
In next PR:

client need to specify the property (both property in key and value) **explicitly**, property in keys will no longer return by default. If no property is specified, all properties in key and value will be returned. This changes will effect all interfaces, including `go/fetch/lookup/scan`. @CPWstatic @yixinglu @bright-starry-sky @Nicole00 

Next PR will not change the thrift interface, but the behavior will be affected. We can do this in December. Query engine, storage and client(scan) need to modify according to it.

### CODE REVIEW

File: src/common/interface/storage.thrift [hunk 1]
```diff
@@ -351,7 +351,7 @@ struct GetPropResponse {
     //
     // Each column represents one peoperty. When all properties are returned, the
     //   column name is in the form of
-    //   "<tag_name>:<prop_alias>" or "<edge_type_name>:<prop_alias>"
+    //   "tag_name.prop_alias" or "edge_type_name.prop_alias"
```
Comment (yixinglu) [inline src/common/interface/storage.thrift:354]: Why not use the same struct definition with LookupResponse ?

Comment (critical27) [reply src/common/interface/storage.thrift:354]: You mean only use one of `GetPropResponse` and `LookupIndexResp`? Not a big difference, bug `GetProp` need to handle `fetch *`.

File: src/common/interface/storage.thrift [hunk 2]
```diff
@@ -614,7 +614,7 @@ struct ScanVertexRequest {
     2: common.PartitionID                   part_id,
     // start key of this block
     3: optional binary                      cursor,
-    4: list<VertexProp>                     return_columns,
+    4: VertexProp                           return_columns,
```
Comment (darionyaphet) [inline src/common/interface/storage.thrift:617]: 👍

File: src/common/interface/storage.thrift [hunk 3]
```diff
@@ -623,6 +623,12 @@ struct ScanVertexRequest {
     // only return data in time range [start_time, end_time)
     7: optional i64                         start_time,
     8: optional i64                         end_time,
+    9: optional binary                      filter,
+    // when storage enable multi versions and only_latest_version is true, only return latest version.
+    // when storage disable multi versions, just use the default value.
```
Comment (darionyaphet) [inline src/common/interface/storage.thrift:628]: when multi-version is disabled, only_latest_version is doesn't work？

Comment (critical27) [reply src/common/interface/storage.thrift:628]: It will work, but it will have one more string copy, that is why I said use default value.

Comment (darionyaphet) [reply src/common/interface/storage.thrift:628]: `default value` is for what?

File: src/common/interface/storage.thrift [hunk 4]
```diff
@@ -652,6 +658,12 @@ struct ScanEdgeRequest {
     // only return data in time range [start_time, end_time)
     7: optional i64                         start_time,
     8: optional i64                         end_time,
+    9: optional binary                      filter,
```
Comment (laura-ding) [inline src/common/interface/storage.thrift:661]: how do you want to use the filter? like the expression? please add the comment

Comment (critical27) [reply src/common/interface/storage.thrift:661]: This is a todo, just the expression in nebula

File: src/common/interface/storage.thrift [hunk 5]
```diff
@@ -652,6 +658,12 @@ struct ScanEdgeRequest {
...
+    // when storage enable multi versions and only_latest_version is true, only return latest version.
+    // when storage disable multi versions, just use the default value.
+    10: bool                                only_latest_version = false,
+    // if set to false, forbid follower read
+    11: bool                                enable_read_from_follower = true,
```
Comment (laura-ding) [inline src/common/interface/storage.thrift:666]: if the follower couldn't make the data is the lastest, when will use it?

Comment (critical27) [reply src/common/interface/storage.thrift:666]: The default value could be false. IMO, scan don't always need to read from leader. One of the most important reason is that it is not a consistent snapshot read in storage (for example, we need more than one request even in a part to get all data). Also, for olap case, not always the latest data is needed, they will specify a time range to get the data in that range. That's why the default value is true.

Comment (Shylock-Hg) [reply src/common/interface/storage.thrift:666]: I guess some plugin like text search or OLAP engine may try it.

File: src/common/interface/storage.thrift (hunk 3 above)
Comment (critical27) [reply src/common/interface/storage.thrift:628]: use default value of `only_latest_version`, which is false

File: src/common/interface/storage.thrift (hunk 5 above)
Comment (critical27) [reply src/common/interface/storage.thrift:666]: What do you mean?

File: src/common/interface/storage.thrift (hunk 3 above)
Comment (darionyaphet) [reply src/common/interface/storage.thrift:628]: when multi-version is disabled, `only_latest_version` is true or false is no impact?

Comment (critical27) [reply src/common/interface/storage.thrift:628]: Always work right, but it will hurt performance when you set it to true and multi version is disabled.

**Review by darionyaphet — APPROVED** [review]

**Review by liuyu85cn — APPROVED** [review]
LGTM

File: src/common/interface/storage.thrift (hunk 1 above)
Comment (yixinglu) [reply src/common/interface/storage.thrift:354]: Yeah. Even if GetProp and Lookup have different implementations, they can also return the same response.

Comment (critical27) [reply src/common/interface/storage.thrift:354]: This looks not so important to me. What really matters is that all the `DataSet` in response must be in same format and query engine don't need to distinguish it from each other. After this merged see the next PR.

**Pull request merged**